pandas/io/parsers/readers.py

```diff
@@ -1485,6 +1485,17 @@ def _make_engine(
             "pyarrow": ArrowParserWrapper,
             "python-fwf": FixedWidthFieldParser,
         }
+
+        file_encoding = getattr(f, "encoding", None)
```

Comment:
This check might be better in `_check_file_or_buffer`

Reply:
I like your suggestion, but trying that out the type checking throws errors. None of the annotated classes in `_make_engine` guarantee an `encoding` attribute, so the `getattr` check would be necessary here anyway at least for the type checker.

Reply:
It's OK adding typing ignores as needed